### PR TITLE
Validation refactor pass 2

### PR DESCRIPTION
**Pull request by XVincentX**
This PR comes out because of an attempt of using `These` in Prism, resulting in some inconsistencies that were not making the thing possible:

* `validateBody` is not doing the `fromEither and fromOption` conversion anymore
* Correctly use `NonEmptyArray` where appropriate
* `validateAgainstSchema` function now returns `None` when no validation errors instead of an empty array — making the things a little bit weird.

### CODE REVIEW

File: packages/http/src/validator/validators/utils.ts [hunk 1]
```diff
@@ -1,49 +1,47 @@
...
-        return Object.assign({}, error, { path });
-      });
-    }
-    return [];
-  });
-};
+export const validateAgainstSchema = (value: unknown, schema: JSONSchema, prefix?: string) =>
+  pipe(
+    Option.tryCatch(() => ajv.compile(schema)),
+    Option.chain(validate => {
```
Comment (StefanDywersant) [inline packages/http/src/validator/validators/utils.ts]:
```suggestion
    Option.mapNullable(validate => {
    ...
    return validate.errors;
```

File: packages/http/src/validator/validators/body.ts [hunk 2]
```diff
@@ -5,17 +5,16 @@ import * as Either from 'fp-ts/lib/Either';
 import * as Option from 'fp-ts/lib/Option';
 import { pipe } from 'fp-ts/lib/pipeable';
 import { get } from 'lodash';
-import * as typeIs from 'type-is';
 import { JSONSchema } from '../../types';
 import { body } from '../deserializers';
 import { IHttpValidator } from './types';
 import { validateAgainstSchema } from './utils';
-import { fromArray } from 'fp-ts/lib/NonEmptyArray';
+import { NonEmptyArray, map } from 'fp-ts/lib/NonEmptyArray';
```
Comment (StefanDywersant) [inline packages/http/src/validator/validators/body.ts]:
```suggestion
import * as NonEmptyArray from 'fp-ts/lib/NonEmptyArray';
```

`map` is too common

File: packages/http/src/validator/validators/body.ts [hunk 3]
```diff
@@ -91,42 +85,53 @@ export class HttpBodyValidator implements IHttpValidator<any, IMediaTypeContent>
...
+            Either.mapLeft(diagnostics => applyPrefix(this.prefix, diagnostics))
+          )
+      )
     );
   }
 }
 
-function applyPrefix(prefix: string, diagnostics: IPrismDiagnostic[]): IPrismDiagnostic[] {
-  return diagnostics.map(d => ({ ...d, path: [prefix, ...(d.path || [])] }));
+function applyPrefix(prefix: string, diagnostics: NonEmptyArray<IPrismDiagnostic>): NonEmptyArray<IPrismDiagnostic> {
```
Comment (StefanDywersant) [inline packages/http/src/validator/validators/body.ts]:
```suggestion
function applyPrefix(prefix: string, diagnostics: NonEmptyArray.NonEmptyArray<IPrismDiagnostic>): NonEmptyArray.NonEmptyArray<IPrismDiagnostic> {
```

File: packages/http/src/validator/validators/body.ts [hunk 4]
```diff
@@ -91,42 +85,53 @@ export class HttpBodyValidator implements IHttpValidator<any, IMediaTypeContent>
...
+  return pipe(
+    diagnostics,
+    map(d => ({ ...d, path: [prefix, ...(d.path || [])] }))
```
Comment (StefanDywersant) [inline packages/http/src/validator/validators/body.ts]:
```suggestion
    NonEmptyArray.map(d => ({ ...d, path: [prefix, ...(d.path || [])] }))
```

File: packages/http/src/mocker/callback/callbacks.ts [hunk 5]
```diff
@@ -12,48 +12,42 @@ import { pipe } from 'fp-ts/lib/pipeable';
 import { generate as generateHttpParam } from '../generator/HttpParamGenerator';
 import { validateOutput } from '../../validator';
 import { parseResponse } from '../../utils/parseResponse';
-import withLogger from '../../withLogger';
 import { violationLogger } from '../../utils/logger';
 import { Logger } from 'pino';
 
-export function runCallback({
+export const runCallback = ({
```
[the remark this answers is not in the document]
Comment (XVincentX) [reply packages/http/src/mocker/callback/callbacks.ts]: Effectively nothing, I'll revert it back

**Pull request merged**